### DISCUSSION TITLE
Implementing an Adapter for Akamai EdgeWorkers

### DISCUSSION THREAD

**Discussion by jchiatt** [honojs/hono#3590] [Ideas]
Hi all, 

As I've been looking into building a "write-once, deploy everywhere" style framework for deploying Edge Compute applications to multiple platforms, Hono has been my `#1` prospect to use as a dependency.

That said, I have a hard requirement to support [Akamai EdgeWorkers](https://techdocs.akamai.com/edgeworkers/docs/welcome-to-edgeworkers), which Hono doesn't have support for. Unfortunately the EdgeWorkers platform has a very different request/response model than other platforms, which I assume is why there is yet to be any support in Hono for the platform.

Over the past 18 months I've gained extensive experience with EdgeWorkers building a middleware-driven framework for it. I'd like to explore how to make an Adapter for EdgeWorkers. I'm in the research phase and am currently trying to get a better understanding of how to integrate with Hono and what types of limitations there might be for an EdgeWorkers Adapter.

My immediate question: is there any documentation that deep dives into how the internals of Hono work? I can obviously deep dive into the code myself, just wondering if there's any pre-existing documentation that could jump-start my research here.

Thanks!

**Comment by maou-shonen** [comment]
Compared to hono.js source code, referencing [node server adapter](https://github.com/honojs/node-server) can help you achieve your goal faster

**Reply by jchiatt** [reply]
Thanks, I'll read through this.

**Comment by erzr** [comment]
@jchiatt are you actively working on Akamai Edgeworkers support? i have some interest in seeing if this is possible as well.
